src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import javafx.application.Application;
+import javafx.event.Event;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+
+import duke.parser.*;
+import duke.storage.Storage;
+import duke.task.*;
+import duke.ui.*;
+
+import java.io.File;
+import java.net.URL;
+import java.util.Scanner;
+
+/**
+ * Main UI method.
+ */
+public class Duke extends Application {
+    public Button sendButton;
+    public ScrollPane scrollPane;
+    public AnchorPane mainLayout;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+
+    public Storage storage;
+    public TaskList tasks;
+    public UiText ui;
+
+    /**
+     * If no file path specified, default path is assumed
+     */
+    public Duke() {
+        this(true);
+    }
+
+    private Duke(boolean gui) {
+        this(Storage.DEFAULT_PATH, gui);
+    }
+
+    /**
+     * constructor to specify file path of the last saved data
+     *
+     * @param filePath = path of last saved data file
+     * @param gui      = using gui?
+     */
+    public Duke(String filePath, boolean gui) {
+        this.ui = new Ui(new Scanner(System.in));
+        this.storage = new Storage(filePath);
+        if (this.storage.fileExist()) {
+            try {
+                this.tasks = TaskList.fromCSVList(storage.loadCSVList());
+            } catch (Exception e) {
+                this.ui.respond(Ui.loadingErrorMsg);
+                tasks = new TaskList();
+            }
+        } else {
+            tasks = new TaskList();
+        }
+        /*
+        if (gui) {
+            this.gui = new Gui(this.dialogContainer, this.userInput);
+        }*/
+    }
+
+    /**
+     * Main method.
```

Comment:
Please follow the conventions for javadoc comment. The comment should start with a simple present e.g. runs xxxx. Please fix others as well.